src/commands.ts

```diff
@@ -91,6 +92,139 @@ export async function updateDependencies(ctx: WorkspaceContext) {
+    const availableProjectTemplates = await ctx.toolchain.getProjectTemplates();
+    const selectedProjectTemplate = await vscode.window.showQuickPick<
+        vscode.QuickPickItem & { type: SwiftProjectTemplate }
+    >(
+        availableProjectTemplates.map(type => ({
+            label: type.name,
+            description: type.id,
+            detail: type.description,
+            type,
+        })),
+        {
+            placeHolder: "Select a swift project template",
+        }
+    );
+    if (!selectedProjectTemplate) {
+        return undefined;
+    }
+    const projectType = selectedProjectTemplate.type.id;
+
+    // Prompt the user for a location in which to create the new project
+    const selectedFolder = await vscode.window.showOpenDialog({
+        title: "Select a folder to create a new swift project in",
+        openLabel: "Select folder",
+        canSelectFiles: false,
+        canSelectFolders: true,
+        canSelectMany: false,
+    });
+    if (!selectedFolder || selectedFolder.length === 0) {
+        return undefined;
+    }
+
+    // Prompt the user for the project name
+    const existingNames = await fs.readdir(selectedFolder[0].fsPath, { encoding: "utf-8" });
+    let initialValue = `swift-${projectType}`;
+    for (let i = 1; ; i++) {
+        if (!existingNames.includes(initialValue)) {
+            break;
+        }
+        initialValue = `swift-${projectType}-${i}`;
+    }
+    const projectName = await vscode.window.showInputBox({
+        value: initialValue,
+        prompt: "Enter a name for your new swift project",
+        validateInput(value) {
+            // Ensure there are no name collisions
+            if (existingNames.includes(value)) {
+                return "A file/folder with this name already exists";
+            }
+            return undefined;
+        },
+    });
+    if (!projectName) {
+        return undefined;
+    }
+
+    // Create the folder that will store the new project
+    const projectUri = vscode.Uri.joinPath(selectedFolder[0], projectName);
+    await fs.mkdir(projectUri.fsPath);
+
+    // Use swift package manager to initialize the swift project
+    await withDelayedProgress(
+        {
+            location: vscode.ProgressLocation.Notification,
+            title: `Creating swift project ${projectName}`,
+        },
+        async () => {
+            await execSwift(
+                ["package", "init", "--type", projectType, "--name", projectName],
+                ctx.toolchain,
+                {
+                    cwd: projectUri.fsPath,
+                }
+            );
+        },
+        1000
+    );
+
+    // Prompt the user whether or not they want to open the newly created project
+    const isWorkspaceOpened = !!vscode.workspace.workspaceFolders;
+    const config = vscode.workspace.getConfiguration("swift");
+    const openAfterCreate = config.get<string>("openAfterCreateProject");
```

Comment:
Can we add configuration settings to configuration.ts

Reply:
Done.